provisioning/transport/amqp/src/amqp.ts

```diff
@@ -312,8 +319,14 @@ export class Amqp extends EventEmitter implements X509ProvisioningTransport, Tpm
             this._senderLink.send(requestMessage, (err) => {
               if (err) {
                 delete this._operations[requestMessage.correlation_id];
-                /*Codes_SRS_NODE_PROVISIONING_AMQP_16_021: [The `queryOperationStatus` method shall call its callback with an error if the transport fails to send the request message.]*/
-                callback(err);
+                const translatedError = translateError('registration failure', err);
+                if ((translatedError instanceof errors.InternalServerError) || ((translatedError as AmqpTransportError) instanceof errors.ThrottlingError)) {
+                  /*Codes_SRS_NODE_PROVISIONING_AMQP_06_002: [ The `queryOperationStatus` disposition is an `instanceof` common errors InternalServerError or ThrottlingError, the result.status shall be set to 'assigning', and the callback invoked with *no* error. ] */
+                  callback(null, {status: 'assigning'});
+                } else {
+                  /*Codes_SRS_NODE_PROVISIONING_AMQP_16_021: [The `queryOperationStatus` method shall call its callback with an error if the transport fails to send the request message.]*/
+                  callback(err);
+                  }
```

Comment:
>   } [](start = 15, length = 4)

nit: indentation

Reply:
Fixed.

---
In reply to: [234754603](https://github.com/Azure/azure-iot-sdk-node/pull/430#discussion_r234754603) [](ancestors = 234754603)